### DISCUSSION TITLE
Ideas Collection | 意见收集帖🧠

### DISCUSSION THREAD

**Discussion by scarsu** [scarsu/NotionX#4] [Ideas]
All suggestions for improving the Notion experience are welcome!
欢迎提出你在使用Notion过程中的痛点，本项目开发者会抓住一切提升Notion使用体验的机会~

**Comment by shirobachi** [comment]
I <3 this extension. May you can add refresh preventer? You know if try to refresh then alert "Are you sure" or sth like that. 
Ctrl + R is so close to Ctrl + E what is useful shortcut :)

**Reply by scarsu** [reply]
Thank you for your idea. I will work on it

**Reply by shirobachi** [reply]
Also I just now think if would be possible to add table of content (if exist) on the right, the best if would be possible to hide it (maybe like left sidebar)

**Reply by shirobachi** [reply]
So I just saw sidebar on some page, but not everyone, look here: https://i.imgur.com/yxonFWs.png
When this sidebar is not available? Like you can see I have no sidebar, I have table of content (w/ content) and have no arrow to appear sidebar. 
Thanks

**Reply by scarsu** [reply]
You can try this : turn off the option "Hide NotionX SideBar" in NotionX extension's popup,and refresh page

**Reply by shirobachi** [reply]
Yes, it works! Thank you! <3

**Reply by scarsu** [reply]
Hi, Shirobachi :)
For your "refresh preventer" idea, I find that if you have content to be saved in your notion page,the broswer will prevent you from refreshing the page,please chekcout this screenshot
![image](https://user-images.githubusercontent.com/19526406/110721091-b7eebf80-824a-11eb-9948-01595d4a494d.png)

**Reply by shirobachi** [reply]
I missed that commend ;/ 

Thanks for the info ofc it's but sometimes it saves too fast :D

**Comment by xukai92** [comment]
插件很好用，非常感谢！请问有没有办法自定义 code hightlighting（如改变 hightlighting 和增加语言）？谢谢！

**Reply by shirobachi** [reply]
He's appreciate for extension and ask if is possible to change highlight syntax code like add new support languages in code block.

**Reply by scarsu** [reply]
感谢你的鼓励和意见~
能力有限，短期内无法增加"自定义hightlighting 和增加语言“功能，很抱歉。

**Reply by scarsu** [reply]
@Shirobachi   
Thank you for your translation :) Actually,chinese is my native language

**Comment by cyanxiao** [comment]
Really love the extension. One suggestion: could you change the style of the toggle button that is used to switch to dark mode? It's not a Notion-style button because of its shape and shadow. Very appreciate it if the toggle button can be redesigned to suit the ones in the `Style` menu. Thank you!

**Reply by scarsu** [reply]
I think you're right.I'll work on it.Thanks a lot.

**Reply by scarsu** [reply]
It's done.Please checkout the latest version.

**Comment by agile-dynamics** [comment]
the #1 feature i need in Notion is this;

in BOARD views, to be able to automatically HIDE any columns that have ZERO cards.

why?

I have many tables in my 'apps' where there are 'multi-select' and 'single-select' properties that are used for TAGS.
Users create new TAGS as required - so the number of unique tags can be large.

But in a BOARD view the system displays ALL the tags across the top - far too many to fit on the screen, or even several screens.
Yet, because of the use of FILTERS in the view, most of these columns are EMPTY. A kind of SPARSE MATRIX.

So I need a way to simply hide all those columns in the board-view that are empty, and only show the columns that have records.
This would be such a powerful tool, not just for MY clients, but for EVERYONE!

BOARD views are an extremely powerful way to present information, and allow users to manipulate it intuitively. This feature would make them very powerful indeed by only showing the minimum number of columns needed, yet allow users to un-hide them if needed.

I dont know enough about building EXTENTIONS to encode this feature myself - so i humbly ask you to consider adding it to NotionX


respect
max

**Reply by agile-dynamics** [reply]
NotionX is a great extention and i really appreciate it - many thanks

so i was wondering if it was possible to implement the above suggestion as part of the extention?

even if the columns of the board-view could be ordered, such that the 'empty' columns occur last (to the right), then the effect would be the same, the user would see the columns that have cards, and the empty columns would be out-of-sight

respect
max

**Reply by scarsu** [reply]
It's a great idea.Thank you.

I'll work on it in next version v2.1.1 before the middle of next month.

**Comment by mohuishou** [comment]
1. 希望右侧的 content 可以只展示浮动目录，并且可以随着页面滚动高亮当前 header
2. 希望可以嵌入到页面当中，而不是一个独立的区域，这样会更美观一些，这部分建议可以参考 wolai 的设计

**Reply by scarsu** [reply]
1. 对于"**只展示浮动目录**"，下一版本正在开发，支持关闭侧边栏动态的HEADERS, TOOGLE, DATABASE, COMMENTS, COLORED TEXT等模块
2. “**随着页面滚动高亮当前 header**” 将纳入开发计划
3. 对于你的**第2点**建议，暂时不予采纳。因为NotionX 的UI参考、尊重并还原了 Notion 原生的左侧侧边栏样式和功能。相比于“嵌入到页面中”，侧边栏的形式功能性更强。

**Comment by vfolee** [comment]
发现了个小问题。
对于太长的代码会选择“Wrap code”让其换行，但是如果配合“配置”里的“显示代码行号”，会导致换行后的代码也被打上行号，如下图
![image](https://user-images.githubusercontent.com/18752088/115171073-ca370580-a0f4-11eb-8e17-b6bf8401342b.png)

对于换行后的代码，能否不分配单独的行号。如下所示
![image](https://user-images.githubusercontent.com/18752088/115171538-cd7ec100-a0f5-11eb-8b51-19a3b01bb415.png)

**Reply by scarsu** [reply]
收到

**Reply by scarsu** [reply]
感谢反馈

**Comment by xiang0731** [comment]
![image](https://user-images.githubusercontent.com/24264988/130383874-9c81bd5c-7b59-443c-bd30-4280347ac255.png)
Hi，发现Notion X 侧边栏常驻的话会直接把后面的按钮挡住，如上图

**Reply by scarsu** [reply]
Hi，感谢反馈，很抱歉带来不好的体验，此问题将在下一版本中修复(预计于2021.9月上旬发布)。

**Reply by five-five0909** [reply]
> Hi，感谢反馈，很抱歉带来不好的体验，此问题将在下一版本中修复(预计于2021.9月上旬发布)。

请问作者 这个项目可以保存为一个js文件，然后供桌面端使用吗，类似于notionCN.js中文的使用

**Comment by tianyunperfect** [comment]
希望把常用的功能放在最上面，可以直接点击，比如为header添加序列号的功能，每次都要滚动下去找在什么位置，不太方便，应该把配置项放在后面，点击项放在上面。
![image](https://user-images.githubusercontent.com/23491291/130915060-cd774c75-0a2e-4ac8-96e9-02ef814edd07.png)

**Comment by tianyunperfect** [comment]
请问谷歌浏览器跨浏览器不能同步设置，这个是我个人的问题吗，其他插件都能同步的。

**Comment by hitopo** [comment]
希望能添加更换正文和代码块字体的功能

**Comment by hepari** [comment]
Thank you so much for this extension. The recently added Auto-numbering for headers was one of the features Notion has been sorely missing.

I have one important request that would really improve Notion: The "Find and Replace" feature.
I don't need anything elegant. I simply need a way to 'find' a certain word on a page and replace all occurrences with another word.

Would this be possible?

**Comment by saruat** [comment]
Hello, can you implement the ability to conduct periodic and one-time tasks in one planner?
Namely, if a one-time task with a specific date is entered into the database, then it is displayed in the calendar 1 time, and if it is recurring, then, depending on the repetition period (daily, weekly, monthly, quarterly, annually, once in "n" days / weeks / months / years), it repeats in the calendar accordingly

**Comment by eisinazuma** [comment]
NotionX is really amazing. I was wondering if you could add a feature in which a header can be a toggle. Being able to hide all the content under a header would be great.

**Comment by Qpp0** [comment]
**设置代码块语言**可否和**生成header序号**一样，点击就批量设置？
现在的问题是：如果我需要代码块都是java，那么如果剪切过来的页面中的代码不是想要的，需要随便切换一种语言，插件批量设置好了再切换成java。
希望：下拉选项设置好想要的语言，然后点击即可批量设置

**Comment by alongnice** [comment]
Regarding the export function, I think it is necessary to automatically add subpages, but the official settings need to open higher permissions. It's a bit painful

**Comment by Swikarneupane** [comment]
Hey love this extension. Would love it if we could use this extension and access our todo list by clicking extension.

**Reply by scarsu** [reply]
Get your idea which is great🥳，I'll be working on that this weekeng.

**Comment by adaliszk** [comment]
I was happy to find the extension, although the one thing I was looking for does not have: Hiding the page title, fields, icon, or cover for a focus mode. Is that possible to add?

**Reply by scarsu** [reply]
Your idea is great.I'll be working on that.

**Comment by Foolisher** [comment]
标题超过10个后，如果重复生成，大于等于10的标题会被重新生成。
标题生成这个功能太棒了，对于一篇文章标题多，层次深的时候，这个功能非常重要，非常期望您修复这个问题啊

**Reply by scarsu** [reply]
很抱歉开发了bug😪，我这周末会抽时间修复，届时在此回复你。

**Reply by Foolisher** [reply]
再提一个的话是，隐藏侧边栏、隐藏暗黑模式的按钮，不生效，要重复点击才能生效。

**Comment by Foolisher** [comment]
大佬~ 超过10个的标题生成还是没有修复呢
